api/src/services/authentication.ts

```diff
@@ -429,6 +441,48 @@ export class AuthenticationService {
 		};
 	}
 
+	private async updateStatefulSession(
+		sessionRecord: Record<string, any>,
+		oldSessionToken: string,
+		newSessionToken: string,
+		sessionExpiration: Date,
+	) {
+		if (sessionRecord['session_next_token']) {
```

Comment:
newSessionToken will indeed be `sessionRecord['session_next_token']` when it is set but it will also contain a [random new session token](https://github.com/directus/directus/pull/22503/files#diff-03b58901729f938551b47a9357d79ae1d0c62868265a05e8f9980d30b22c0e95R361) if that column is not set in the database. Using `newSessionToken` here would always resolve to `true` as it will always have a value.

Reply:
I don't think so, `newSessionToken` always exists.

https://github.com/directus/directus/blob/ada0a51ecc5bef5ff2b142af9f7b16bded0918a2/api/src/services/authentication.ts#L361

Reply:
Thank you for the explanation.
It makes sense now 🙏